h2o-core/src/main/java/water/fvec/NewChunk.java

```diff
@@ -648,16 +648,21 @@ public void addUUID(UUID uuid) {
 
   // Append a UUID, stored in _ls & _ds
   public void addUUID( long lo, long hi ) {
-    if (C16Chunk.isNA(lo, hi)) throw new IllegalArgumentException("Cannot set illegal UUID value");
-    if( _ms==null || _ds== null || _sparseLen >= _ms.len() )
-      append2slowUUID();
-    _ms.set(_sparseLen,lo);
-    _ds[_sparseLen] = Double.longBitsToDouble(hi);
-    if (_id != null) _id[_sparseLen] = _len;
+    set_impl(_sparseLen, lo, hi);
     _sparseLen++;
     _len++;
     assert _sparseLen <= _len;
   }
+
+  @Override public boolean set_impl(int idx, long lo, long hi) {
```

Comment:
I agree with Michal. 

Set should not be growing chunk. Interface in our chunks set is RA set, must be within bounds of 0,_len, *does not* insert after _len. This is how Vecs are designed.